articles/ai-services/openai/how-to/use-web-app.md

```diff
@@ -45,7 +63,331 @@ When you're customizing the app, we recommend:
+```json
+  {
+    "name": "UI_CHAT_DESCRIPTION",
+    "value": "This is an example of a UI Chat Description. Chatbots can make mistakes. Check important info and sensitive info.",
+    "slotSetting": false
+  },
+  {
+    "name": "UI_CHAT_LOGO",
+    "value": "https://learn-bot.azurewebsites.net/assets/Contoso-ff70ad88.svg",
+    "slotSetting": false
+  },
+  {
+    "name": "UI_CHAT_TITLE",
+    "value": "This is an example of a UI Chat Title. Start chatting",
+    "slotSetting": false
+  },
+  {
+    "name": "UI_FAVICON",
+    "value": "https://learn-bot.azurewebsites.net/assets/Contoso-ff70ad88.svg",
+    "slotSetting": false
+  },
+  {
+    "name": "UI_LOGO",
+    "value": "https://learn-bot.azurewebsites.net/assets/Contoso-ff70ad88.svg",
+    "slotSetting": false
+  },
+  {
+    "name": "UI_TITLE",
+    "value": "This is an example of a UI Title",
+    "slotSetting": false
+  },
+```
+
+## Enabling chat history using Cosmos DB
+
+You can turn on chat history for your users of the web app. When you turn on the feature, users have access to their individual previous queries and responses.
+
+To turn on chat history, deploy or redeploy your model as a web app by using [Azure OpenAI Studio](https://oai.azure.com/portal) or [Azure AI Studio](https://ai.azure.com/) and select **Enable chat history and user feedback in the web app**.
+
+:::image type="content" source="../media/use-your-data/enable-chat-history.png" alt-text="Screenshot of the checkbox for enabling chat history in Azure OpenAI or Azure AI Studio." lightbox="../media/use-your-data/enable-chat-history.png":::
+
+> [!IMPORTANT]
+> Turning on chat history creates an [Azure Cosmos DB](/azure/cosmos-db/introduction) instance in your resource group, and it incurs [additional charges](https://azure.microsoft.com/pricing/details/cosmos-db/autoscale-provisioned/) for the storage that you use beyond any free tiers.
+
+After you turn on chat history, your users can show and hide it in the upper-right corner of the app. When users show chat history, they can rename or delete conversations. You can modify whether users can access this function using the environment variable `UI_SHOW_CHAT_HISTORY_BUTTON` as specified in the previous section. Because the users are signed in to the app, conversations are automatically ordered from newest to oldest. Conversations are named based on the first query in the conversation.
+
+Note that popular Azure regions such as East US may experience periods of high-demand where it may not be possible to provision a new instance of Cosmos DB. In that case, opt to deploy to alternative region such as East US 2 or retry your deployment until it succeeds. Should the deployment of Cosmos DB fail, your app will be available at its specified URL, but chat history will not be available. Enabling conversation history will also enable the view conversation history button in the top-right.
```

Comment:
```suggestion
> [!Note]
> Popular Azure regions such as East US may experience periods of high-demand where it may not be possible to provision a new instance of Cosmos DB. In that case, opt to deploy to alternative region such as East US 2 or retry your deployment until it succeeds. Should the deployment of Cosmos DB fail, your app will be available at its specified URL, but chat history will not be available. Enabling conversation history will also enable the view conversation history button in the top-right.
```